src/Population.jl

```diff
@@ -167,7 +167,7 @@ const CACHED_WEIGHTS =
         PerThreadCache{Dict{Tuple{Int,Float32},typeof(test_weights)}}()
     end
 
-@unstable function get_tournament_selection_weights(@nospecialize(options::Options))
+@unstable function get_tournament_selection_weights(@nospecialize(options::AbstractOptions))
     n = options.tournament_selection_n
     p = options.tournament_selection_p
```

Comment:
```suggestion
    n = options.tournament_selection_n::Int
    p = options.tournament_selection_p::Float32
```

But it likely doesn't affect things that much. I think the original reason I limited this to `Options` rather than `AbstractOptions` was because having the `@nospecialize` would result in the compiler being unable to infer the type of each of these. But just having these assertions there will probably help.